src/partners/0xgasless.ts

```diff
@@ -0,0 +1,250 @@
+            '0x-version': 'v2'
+          }
+        })
+        const responseJson = await response.text()
+        if (!response.ok) {
+          throw new Error(`${url} response ${response.status}: ${responseJson}`)
+        }
+        const responseBody = asGetGaslessTradesResponse(responseJson)
+
+        for (const trade of responseBody.trades) {
+          const buySymbol = trade.tokens.find(t => t.address === trade.buyToken)
+            ?.symbol
+          const sellSymbol = trade.tokens.find(
+            t => t.address === trade.sellToken
+          )?.symbol
+
+          if (buySymbol == null || sellSymbol == null) {
+            throw new Error(
+              `Could not find buy or sell symbol for trade ${trade.zid}; txid: ${trade.transactionHash}`
+            )
+          }
+
+          const {
+            isoDate: tradeIsoDate,
+            timestamp: tradeTimestamp
+          } = smartIsoDateFromTimestamp(trade.timestamp * 1000)
+
+          // If trade is 2 days or older, then it's finalized according to 0x
+          // documentation.
+          const status: Status =
+            tradeTimestamp + 2 * 24 * 60 * 60 * 1000 < now
+              ? 'complete'
+              : 'pending'
+
+          const ssTx: StandardTx = {
+            status,
+            orderId: trade.zid,
+            depositTxid: trade.transactionHash,
+            depositAddress: undefined,
+            depositCurrency: sellSymbol,
+            depositAmount: Number(trade.sellAmount),
+            payoutTxid: trade.transactionHash,
+            payoutAddress: trade.taker ?? undefined,
+            payoutCurrency: buySymbol,
+            payoutAmount: Number(trade.buyAmount),
+            timestamp: tradeTimestamp,
+            isoDate: tradeIsoDate,
+            usdValue: parseFloat(trade.volumeUsd),
+            rawTx: trade
+          }
+          ssFormatTxs.push(ssTx)
+          if (ssTx.isoDate > latestIsoDate) {
+            latestIsoDate = ssTx.isoDate
+          }
+        }
+
+        datelog(`0xGasless ${responseBody.trades.length} trades processed`)
+
+        if (responseBody.nextCursor == null) {
+          datelog(`0xGasless No cursor from API`)
+          break
+        } else {
+          cursor = responseBody.nextCursor
+          datelog(`0xGasless Get nextCursor: ${cursor}`)
+        }
+      }
+
+      lastCheckedTimestamp = endTimestamp
+      datelog(
+        `0xGasless endDate:${new Date(
+          lastCheckedTimestamp
+        ).toISOString()} latestIsoDate:${latestIsoDate}`
+      )
+      if (lastCheckedTimestamp > now) {
+        break
+      }
+      retry = 0
+    } catch (error) {
+      datelog(error)
+      // Retry a few times with time delay to prevent throttling
+      retry++
+      if (retry <= MAX_RETRIES) {
+        datelog(`Snoozing ${60 * retry}s`)
+        await snooze(60000 * retry)
+      } else {
+        // We can't safely save our progress since we go from newest to oldest.
```

Comment:
technically not true since you use a date range this is earlier and work your way up to present. One small fix that would be needed (and also in the paybis plugin) is to only set `latestIsoDate` if you have fully finished

Reply:
See my untested fixup commit